transit/script/bart.py

```diff
@@ -0,0 +1,396 @@
+
+    return BART_SL
+
+
+station_locations = {
+    "downtown": ["CIVC", "POWL", "MONT", "EMBR"],
+    "not_downtown": ["GLEN", "BALB", "24TH", "16TH"],
+}
+
+# Create station to screenline mapping
+station_to_label = {}
+for loc in station_locations:
+    stations = station_locations[loc]
+    labels = [loc] * len(station_locations[loc])
+    station_to_label = station_to_label | dict(zip(stations, labels))
+
+relevant_stations = station_locations["downtown"] + station_locations["not_downtown"]
+
+
+def determineSL(x):
+    if (x["A_SL"] == "not_downtown") & (x["B_SL"] == "downtown"):
+        return "IB"
+    elif (x["A_SL"] == "downtown") & (x["B_SL"] == "not_downtown"):
+        return "OB"
+    else:
+        return False
+
+
+def process_BART_SF(filename, model_run_dir):
+    nodes = read_nodes(model_run_dir)
+    df_station_name = station_name()
+    lines = ["BART", "EBART", "OAC"]
+
+    # Read, group, and sum data for each line
+    dfs = [
+        read_dbf_and_groupby_sum(filename, line, ["A", "B", "TOD"], "AB_VOL")
+        for line in lines
+    ]
+
+    # Concatenate results
+    intra = pd.concat(dfs)
+
+    # Mapping from Node and Station DataFrames
+    node_mapping = nodes.set_index("Node")["Node Name"].to_dict()
+    station_mapping = df_station_name.set_index("Node")["Station"].to_dict()
+
+    # Apply mappings
+    intra["A_name"] = intra["A"].map(node_mapping)
+    intra["B_name"] = intra["B"].map(node_mapping)
+    intra["A_station"] = intra["A"].map(station_mapping)
+    intra["B_station"] = intra["B"].map(station_mapping)
+
+    # Filter out specific values and stations
+    values_to_drop = ["Hillcrest eBART", "Coliseium OAC", "Somersville Road eBART"]
+    intra = intra[
+        ~intra["A_name"].isin(values_to_drop) & ~intra["B_name"].isin(values_to_drop)
+    ]
+    intra = intra[
+        intra["A_station"].isin(relevant_stations)
+        & intra["B_station"].isin(relevant_stations)
+    ]
+
+    # Apply custom station labeling and direction determination
+    intra["A_SL"] = intra["A_station"].apply(lambda x: station_to_label[x])
+    intra["B_SL"] = intra["B_station"].apply(lambda x: station_to_label[x])
+    intra["Direction"] = intra.apply(lambda x: determineSL(x), axis=1)
+    intra = intra[intra["Direction"] != False]
+
+    # Set time of day and select final columns
+    intra = intra[["Direction", "TOD", "AB_VOL"]]
+    # 2019 validation uses 'Intra-SF'
+    intra["Screenline"] = "SF-San Mateo"
+    intra["Key"] = intra["Screenline"] + intra["Direction"] + intra["TOD"]
+    intra.columns = ["Direction", "TOD", "Ridership", "Screenline", "Key"]
+    intra = intra[["Screenline", "Direction", "TOD", "Key", "Ridership"]]
+    intra = intra.sort_values(by=["Direction", "TOD"]).reset_index(drop=True)
+
+    return intra
+
+
+def process_BART_SL(file_name, model_run_dir, output_dir, model_BART_SL):
+    transbay_node = [16510, 16511]  # 16510 in, 16511 out
```

Comment:
hmm should node numbers be in another toml file or a external data table? @sdrewc @dantischler thoughts?

Reply:
toml file. Not necessary to create another file just to store 2 numbers

Reply:
@chunhochow and @ykj5060 I see a lot of pretty minor issues in here.  Can we identify which ones are top priority for getting user-friendly, working code?  I'd like to have a release we can use this week, ideally.

Reply:
well there's a whole list of BART nodes after these two
@sdrewc I'll take a look at all my comments again and flag the critical ones. for the others, we can always just merge to master first (once @ykj5060 has run everything to make sure that it runs as is) and then tackle these afterwards